packages/fast-usdc/src/clientSupport.js

```diff
@@ -0,0 +1,110 @@
+// @ts-check
+import { AmountMath } from '@agoric/ertp';
+import { assertAllDefined } from '@agoric/internal';
+
+// Reusable amount scaling similar to inter-protocol's approach
+const COSMOS_UNIT = 1_000_000n;
+const scaleDecimals = num => BigInt(num * Number(COSMOS_UNIT));
+
+/**
+ * @param {Pick<
+ *   import('@agoric/vats/tools/board-utils.js').AgoricNamesRemotes,
+ *   'brand'
+ * >} agoricNames
+ * @param {Instance} instance
+ * @param {object} opts
+ * @param {string} opts.offerId
+ * @param {number} opts.amount
+ * @returns {import('@agoric/smart-wallet/src/offers.js').OfferSpec}
+ */
+const makeDepositOffer = ({ brand }, instance, { offerId, amount }) => {
+  assertAllDefined({ offerId, amount });
+  
+  return {
+    id: offerId,
+    invitationSpec: {
+      source: 'contract',
+      instance,
+      publicInvitationMaker: 'makeDepositInvitation',
+    },
+    proposal: {
+      give: {
+        Deposit: {
+          brand: brand.USDC,
+          value: scaleDecimals(amount)
+        }
+      }
+    }
+  };
+};
+
+/**
+ * @param {Pick<
+ *   import('@agoric/vats/tools/board-utils.js').AgoricNamesRemotes,
+ *   'brand'
+ * >} agoricNames
+ * @param {Instance} instance
+ * @param {object} opts
+ * @param {string} opts.offerId
+ * @param {number} opts.amount
+ * @returns {import('@agoric/smart-wallet/src/offers.js').OfferSpec}
+ */
+const makeWithdrawOffer = ({ brand }, instance, { offerId, amount }) => ({
+  id: offerId,
+  invitationSpec: {
+    source: 'contract',
+    instance,
+    publicInvitationMaker: 'makeWithdrawInvitation',
+  },
+  proposal: {
+    want: {
+      Withdrawal: {
+        brand: brand.USDC,
+        value: scaleDecimals(amount)
+      }
+    }
+  }
+});
+
+/**
+ * @param {Instance} instance
+ * @param {object} opts
+ * @param {string} opts.offerId
+ * @param {string} opts.operatorId
+ * @returns {import('@agoric/smart-wallet/src/offers.js').OfferSpec}
+ */
+const makeOperatorOffer = (instance, { offerId, operatorId }) => ({
+  id: offerId,
+  invitationSpec: {
+    source: 'contract',
+    instance,
+    callPipe: [['makeOperatorInvitation', [operatorId]]]
+  },
+  proposal: {}
+});
+
+/**
+ * @param {Instance} instance
+ * @param {object} opts
+ * @param {string} opts.offerId
+ * @returns {import('@agoric/smart-wallet/src/offers.js').OfferSpec}
+ */
+const makeWithdrawFeesOffer = (instance, { offerId }) => ({
```

Comment:
This one is much less valuable; only a core-eval can withdraw fees.

But if we want to provide a helper, let's factor it out of...

https://github.com/Agoric/agoric-sdk/blob/f0ccde6550d7f0d22fd489205cb12a6598faa2c9/packages/fast-usdc/src/distribute-fees.core.js#L32

Note support for both absolute amounts and proportions.